src/plot_api/plot_api.js

```diff
@@ -2219,7 +2219,7 @@ Plotly.animate = function(gd, frameOrGroupNameOrFrameList, animationOpts) {
 
                 var nextFrame = {
                     frame: computedFrame,
-                    name: frameList[i].name,
+                    name: frameList[i].name || computedFrame.name,
```

Comment:
That's not a bad point. I can move that elsewhere (being careful not to mutate the user input itself) for this particular case. (It's the case when a plain object is provided instead of the name of a frame.)